qt-bugs@ issue : none
Trolltech task ID : none
bugs.kde.org number : none
applied: yes
author: Lubos Lunak <l.lunak@kde.org>

This patch changes QObjectPrivateVersion, thus preventing mixing
parts of upstream Qt and qt-copy. In general it is a bad idea to mix
e.g. libQtCore from one build and libQtGui from another one, and other
qt-copy patches could make changes in Qt internal structures that could
cause problems when mixed with upstream Qt.

This patch does not make qt-copy binary incompatible with upstream Qt.
It only further enforces using the same sources for the whole Qt build.

--- src/corelib/kernel/qobject_p.h.sav	2008-01-29 19:37:26.000000000 +0100
+++ src/corelib/kernel/qobject_p.h	2008-01-30 14:08:15.000000000 +0100
@@ -75,7 +75,9 @@ extern QSignalSpyCallbackSet Q_CORE_EXPO
 
 inline QObjectData::~QObjectData() {}
 
-enum { QObjectPrivateVersion = QT_VERSION };
+// add 0x1000000 to mark it as qt-copy version, with possible modifications
+// in some Q*Private class
+enum { QObjectPrivateVersion = QT_VERSION + 0x1000000 };
 
 class Q_CORE_EXPORT QObjectPrivate : public QObjectData
 {
